src/applications/check-in/containers/withSession.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useDispatch, batch } from 'react-redux';
+import propTypes from 'prop-types';
+
+import { createSetSession } from '../actions/authentication';
+import { createInitFormAction } from '../actions/navigation';
+import { setVeteranData, updateFormAction } from '../actions/pre-check-in';
+import { createSetEditContext } from '../actions/edit';
+
+import { useSessionStorage } from '../hooks/useSessionStorage';
+import { useFormRouting } from '../hooks/useFormRouting';
+
+import { URLS } from '../utils/navigation';
+
+import { createForm } from '../utils/navigation/pre-check-in';
+
+import { APP_NAMES } from '../utils/appConstants';
+
+import { api } from '../api';
+
+const withSession = (Component, options = {}) => {
+  const WrappedComponent = props => {
+    const { isPreCheckIn } = options;
+    const { router } = props;
+    const dispatch = useDispatch();
+
+    const { jumpToPage, goToErrorPage } = useFormRouting(router);
+    const { getCurrentToken } = useSessionStorage(isPreCheckIn);
+    const [isLoading, setIsLoading] = useState(true);
+    const [loadingMessage, setLoadingMessage] = useState(
+      'checking for session',
+    );
+
+    const setSession = useCallback(
+      (token, permissions) => {
+        dispatch(createSetSession({ token, permissions }));
+      },
+      [dispatch],
+    );
+
+    const initForm = useCallback(
+      (pages, firstPage) => {
+        dispatch(createInitFormAction({ pages, firstPage }));
+      },
+      [dispatch],
+    );
+
+    const dispatchSetVeteranData = useCallback(
+      payload => {
+        batch(() => {
+          dispatch(setVeteranData({ ...payload }));
+          dispatch(updateFormAction({ ...payload }));
+        });
+      },
+      [dispatch],
+    );
+
+    const setEditContext = useCallback(
+      (data, url) => {
+        dispatch(
+          createSetEditContext({
+            ...data,
+          }),
+        );
+        jumpToPage(url);
+      },
+      [dispatch, jumpToPage],
+    );
+
+    useEffect(
```

Comment:
NOTE: This `useEffect` is what loads the data when a page is loaded. 

When a user logs in we lose all the sessions on the client-side and this rebuilds it. It uses only data in the URL that is built [here](https://github.com/department-of-veterans-affairs/vets-website/pull/20673/files#diff-db47f71922a221fa2b48ae702ce3cdd34e6f0d5cb6b1b5f3fe275049ed10d409R42)

The `if` on line 78ish _may_ need to be refined when this gets past prototype stage.